src/lib.rs

```diff
@@ -1117,14 +1130,27 @@ impl<T: Counter> Histogram<T> {
 
     /// Computes the matching histogram value for the given histogram bin.
     pub fn value_for(&self, index: usize) -> u64 {
+        // TODO what to do about indexes that are beyond what can be represented by this histogram?
+        // Pretty easy to end up shifting off the high end of a u64, e.g. asking for a big index
+        // when sigfigs is high.
```

Comment:
Yeah, this is something I thought about back in the day, but didn't find a nice solution to. I guess the right thing to do is to make `value_for` return a `Result<u64>`, and then use `?` liberally in any users of `value_for`. I worry that this will make *everything* return `Result` though..